### PR TITLE
Remove Travis CI, publish to Coveralls from GH Actions

### PR DESCRIPTION
**Pull request by rock3r**
This PR removes the Travis config, and moves the Coveralls publishing duty to the buildbot GH Action.

Note, we had to switch to a new Coveralls Gradle plugin as the one we were using is [essentially abandoned](kt3k/coveralls-gradle-plugin#107), and while it technically supports GH Actions according to its changelog, it's not documented at all, and I can't figure out how to set it up properly.

### CODE REVIEW

File: .coveralls.yml [hunk 1]
```diff
@@ -1,2 +0,0 @@
-service_name: travis-ci
-repo_token: O7N8dGjWu6Z2BEcBkkfggKgwQecw5lvqQ
```
Comment (rock3r) [inline .coveralls.yml:2]: Note, this is not meant to be public. You should regenerate it from the coveralls.io repo settings!

File: .github/workflows/buildbot.yml [hunk 2]
```diff
@@ -7,17 +7,23 @@ on:
...
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle.kts') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Check with Gradle
+        run: ./gradlew --no-daemon check jacocoTestReport coverallsJacoco
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (rock3r) [inline .github/workflows/buildbot.yml:29]: This should work fine, but may need to be changed to `secrets. COVERALLS_REPO_TOKEN` (see [instructions](https://github.com/nbaztec/coveralls-jacoco-gradle-plugin#github-actions)) if the build fails. Theoretically the `GITHUB_TOKEN` version should allow it to work for forks, too — and it does on my fork.

File: build.gradle.kts [hunk 3]
```diff
@@ -50,7 +58,3 @@ tasks {
         }
     }
 }
-
-coveralls {
```
Comment (rock3r) [inline build.gradle.kts:54]: No longer needed, the new jacoco/coveralls plugin is made for Kotlin projects and thus knows how to pick code up properly.

File: .github/workflows/buildbot.yml (hunk 2 above)
Comment (rock3r) [reply .github/workflows/buildbot.yml:29]: Yup, it works, yay!

File: .coveralls.yml (hunk 1 above)
Comment (mpgirro) [reply .coveralls.yml:2]: Whoops, you're quite right 😳

**Pull request merged**